### ISSUE TITLE
Flask-SocketIO cross-contamination from other app

### ISSUE THREAD

**Issue by kf106** [miguelgrinberg/Flask-SocketIO#882]
I have two applications (bank-app.py and client-app.py), two separate venv installations, one for each. The bank-app.py uses socketio, the client-app.py is a plain flask app.

When I try to run the client-app on a different port, I get two serving messages, and it runs on the wrong port. This only started happening when I added the socketio stuff to the bank-app.py

The code is at https://github.com/kf106/blockchain-games
```
(venv-client) kf106@media-pc:/media/kf106/Elements/GIT/blockchain-games$ FLASK_RUN_PORT=5002 FLASK_APP=client-app.py flask run
 * Serving Flask-SocketIO app "client-app.py"
 * Forcing debug mode off
 * Serving Flask app "client-app.py"
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
```
Any suggestions what is going on here?

**Comment by kf106** [comment]
So further investigation with a clean working directory: a basic flask app with a venv with no socketio works fine:
```
(venv) kf106@media-pc:~/test$ export FLASK_RUN_PORT=6667
(venv) kf106@media-pc:~/test$ FLASK_APP=client-app.py flask run
 * Serving Flask app "client-app.py"
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:6667/ (Press CTRL+C to quit)

```

 but if I delete the venv folder and pycache, add `Flask-SocketIO==3.1.2` to requirements.txt and then run through the same process again, SocketIO turns up, and the changed port (was 6667) is over-riden for the default 5000 port:
```
(venv) kf106@media-pc:~/test$ export FLASK_RUN_PORT=6667
(venv) kf106@media-pc:~/test$ FLASK_APP=client-app.py flask run
 * Serving Flask-SocketIO app "client-app.py"
 * Forcing debug mode off
 * Serving Flask app "client-app.py"
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
```

**Comment by kf106** [comment]
Furthermore, in the second case, if I stop the flask server, go into `/home/kf106/test/venv-client/lib/python3.6` and manually delete the `Flask_SocketIO-3.1.2.dist-info` and `flask-socketio` folders, then run the flask run command again, it produces the first output.

**Comment by kf106** [comment]
I guess this is related to this: https://github.com/miguelgrinberg/Flask-SocketIO/issues/713

**Comment by kf106** [comment]
So as a work-around, how can I get the FLASK_RUN_PORT working again?

**Comment by miguelgrinberg** [comment]
@kf106 based on your description, my suggestion is that you do not install Flask-SocketIO in the venv for the project that doesn't use it.

At some point I had a fix in place for #713, but had to undo it as it affected the use of the reloader. 

My currently thinking is that I'm going to stop supporting the use of `flask run` to start Flask-SocketIO apps. How are you starting your bank-app.py application? Do you use `flask run` or do you go with the "normal" way of running `socketio.run(app)`? I want to require the latter, so that I don't have to intercept `flask run`, which has proven very hard and also a moving target.

**Comment by miguelgrinberg** [comment]
Starting with Flask-SocketIO 3.2.0 (just released), the `flask run` command can only be used with the threading mode. For all other async modes, the `socketio.run(app)` method should be used to start the server.
